tracer/src/Datadog.Trace/IAST/IastModule.cs

```diff
@@ -64,11 +65,28 @@ public IastModule()
 
         if (!iast.Settings.DeduplicationEnabled || HashBasedDeduplication.Instance.Add(vulnerability))
         {
-            // The VulnerabilityBatch class is not very useful right now, but we will need it when handling requests
+            return AddVulnerability(tracer, integrationId, operationName, vulnerability);
+        }
+
+        return null;
+    }
+
+    private static Scope? AddVulnerability(Tracer tracer, IntegrationId integrationId, string operationName, Vulnerability vulnerability)
+    {
+        var iActiveScope = DistributedTracer.Instance.GetActiveScope() ?? tracer.TracerManager?.ScopeManager.Active;
+        var traceContext = (iActiveScope as Scope)?.Span?.Context?.TraceContext;
+
+        if (traceContext?.RootSpan?.Type == SpanTypes.Web)
+        {
+            traceContext?.IastRequestContext.AddVulnerability(vulnerability);
```

Comment:
`Scope.Span` and `Span.Context` cannot be null, and `traceContext` cannot be null inside this `if` condition.
```suggestion
        var traceContext = (iActiveScope as Scope)?.Span.Context.TraceContext;

        if (traceContext?.RootSpan?.Type == SpanTypes.Web)
        {
            traceContext.IastRequestContext.AddVulnerability(vulnerability);
```

Reply:
Or if you want to get fancy with pattern-matching 😅 
```csharp
if (scope is Scope { Span.Context.TraceContext: { RootSpan.Type: SpanTypes.Web } traceContext })
{
    traceContext.IastRequestContext.AddVulnerability(vulnerability);
}
```